src/modules/user/user.service.ts

```
@@ -50,4 +53,44 @@ export class UserService {
   ): Promise<User> => {
     return this.userRepository.createUser(createUserDto, passwordSalt);
   };
+
+  /**
+   * update user
+   * @param updateUserDto - UpdateUserDto that has updated user information
+   * @returns A promise resolving to the updated User object or null
+   */
+  updateUserById = async (
+    updateUser: UpdateUserDto,
+  ): Promise<UserResponseDto> => {
+    // check if the user exists
+    const foundUser = await this.userRepository.getUserById(updateUser.id);
+    if (!foundUser) {
+      throw ServiceException.EntityNotFoundException(
+        errorMessages.ENTITY_NOT_FOUND('User', updateUser.id.toString()),
+      );
+    }
+
+    // check for duplicated emails
+    if (foundUser.email !== updateUser.email) {
+      const isExist = await this.doesUserWithEmailExist(updateUser.email);
+      if (isExist) {
+        throw ServiceException.BadRequestException(
+          `User with email ${updateUser.email} exists already`,
+        );
+      }
+    }
+
+    const passwordInformation =
+      await this.passwordEncryption.createSaltAndHashedPassword(
+        updateUser.password,
+      );
+    updateUser.password = passwordInformation.hashedPassword;
+
+    const updatedUserFromRepo = await this.userRepository.updateUser(
```

Can you also get jwt token user id from request object from controller and compare if updateUserDto.id is same as jwt token user id?
if it's not same, it should throw an error (403 => refer to errorMessages in common folder! Also, please check out how update project api deals with it :-))
This is necessary because another user that is not user himself/herself shouldn't update user.